docs/usage/esql.asciidoc

```diff
@@ -0,0 +1,78 @@
+[[esql]]
+=== ES|QL in the Java client
+
+This page enables you to understand and use {ref}/esql.html[ES|QL] in the
+Java client.
+
+There are two ways to use ES|QL in the {java-client):
+
+* by using the Elasticsearch {es-docs}/esql-apis.html[ES|QL API] directly. This
+is the most flexible approach, but also the most complex since you must handle
+results in their raw form. You can also choose the precise format of results,
+such as JSON, CSV or text.
+* by using ES|QL mapping helpers. These mappers take care of parsing the raw
+response into something readily usable by the application. Several mappers are
+available that are suited to different use cases such as object mapping, cursor
+traversal of results or dataframes. You can also define your own for specific
+use cases.
+
+
+
+[discrete]
+[[esql-how-to]]
+==== How to use the ES|QL API
+
+The {es-docs}/esql-query-api.html[ES|QL query API] allows you to choose how
+results should be returned. This includes the 
+{es-docs}/esql-rest.html#esql-rest-format[response format] such as CSV, text or
+JSON, which can be further fine-tuned with parameters such as column separators
+or locale.
+
+Since the response format varies widely depending on the chosen format, the
+{java-client) as a BinaryData object, which you can then use according to the
+format chosen in the request.
```

Comment:
```suggestion
Because the response varies widely depending on the format, the
{java-client} has a BinaryData object you can use according to the
format specified in the request.
```

Reply:
Seemed like there were typos here:
`{java-client) as`

But I'm not sure this paragraph is clear enough in general yet. I'm sure we can refine it as we go.